packages/host/app/resources/auto-attached-card.ts

```diff
@@ -1,40 +1,86 @@
     attachedCards: CardDef[] | undefined; // cards manually attached in ai panel
   };
 }
 
 /**
- * Manages the auto-attachment of the top-most card in consideration of user-actions of manually
+ * Manages the auto-attachment of cards within our stack in consideration of user-actions of manually
  * removing and attaching new cards in the ai panel
  */
-
 export class AutoAttachment extends Resource<Args> {
-  @tracked card: CardDef | undefined;
-  private lastCard: CardDef | undefined;
+  cards: TrackedSet<CardDef> = new TrackedSet(); // auto-attached cards
+  private lastStackedItems: StackItem[] = [];
+  private lastRemovedCards: Set<string> = new Set(); // internal state, changed from the outside. It tracks, everytime a card is removed in the ai-panel
 
   modify(_positional: never[], named: Args['named']) {
-    const { lastTopMostCard, attachedCards } = named;
-    if (lastTopMostCard === undefined) {
-      this.card = undefined;
-      this.lastCard = undefined;
-      return;
+    const { stackItems, attachedCards } = named;
+    if (this.stackItemsChanged(stackItems)) {
+      // we must be sure to clear the lastRemovedCards state so cards can be auto-attached again
+      // note: if two of the same cards are opened on separate stack, one will be auto-attached.
+      // If one is removed from one of the stacks, the card WILL be auto-attached.
+      this.lastRemovedCards.clear();
+    }
+    this.cards.clear();
+    stackItems.forEach((item) => {
+      if (item === undefined) {
+        // TODO: another place where stackItems = [undefined]. Pls remove after finding root cause
+        return;
+      }
+      if (!this.hasRealmURL(item) || this.isIndexCard(item)) {
+        return;
+      }
+      if (
+        this.isAlreadyAttached(item.card, attachedCards) ||
+        this.wasPreviouslyRemoved(item.card)
+      ) {
+        return;
+      }
+      this.cards.add(item.card);
+    });
+    this.lastStackedItems = stackItems;
+  }
+
+  stackItemsChanged(stackItems: StackItem[]) {
+    if (stackItems.length !== this.lastStackedItems.length) {
+      return true;
     }
-    if (
-      this.isAlreadyAttached(lastTopMostCard, attachedCards) ||
-      this.wasPreviouslyCleared(lastTopMostCard)
-    ) {
-      this.card = undefined;
-      return;
+    for (let i = 0; i < stackItems.length; i++) {
+      if (stackItems[i].card.id !== this.lastStackedItems[i].card.id) {
+        return true;
+      }
     }
-    this.card = lastTopMostCard;
+    return false;
+  }
+
+  private hasRealmURL(stackItem: StackItem) {
+    // if (!('card' in stackItem )) {
+    //   return false;
+    // }
+    let realmURL = stackItem.card[stackItem.api.realmURL];
+    if (!realmURL) {
+      return false;
+    }
+    return true;
+  }
+
+  private isIndexCard(stackItem: StackItem) {
+    let realmURL = stackItem.card[stackItem.api.realmURL];
+    if (stackItem.card.id === `${realmURL!.href}index`) {
+      return true;
+    }
+    return false;
```

Comment:
this is better written as:
```suggestion
return stackItem.card.id === `${realmURL!.href}index`;
```

Reply:
(generally, unless i mean to break out of a loop, whenever I see a `return true;` in the body of an `if` it's usually a code smell)